Assets/MixedRealityToolkit.Tests/PlayModeTests/SolverTests.cs

```diff
@@ -0,0 +1,246 @@
+
+#if !WINDOWS_UWP
+// When the .NET scripting backend is enabled and C# projects are built
+// The assembly that this file is part of is still built for the player,
+// even though the assembly itself is marked as a test assembly (this is not
+// expected because test assemblies should not be included in player builds).
+// Because the .NET backend is deprecated in 2018 and removed in 2019 and this
+// issue will likely persist for 2018, this issue is worked around by wrapping all
+// play mode tests in this check.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.Utilities.Solvers;
+using NUnit.Framework;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    public class SolverTests : BasePlayModeTests
+    {
+        private const float DistanceThreshold = 2.5f;
+        private const float SolverUpdateWaitTime = 2.0f; //seconds
+        /// <summary>
+        /// Internal class used to store data for setup
+        /// </summary>
+        protected class SetupData
+        {
+            public SolverHandler handler;
+            public Solver solver;
+            public GameObject target;
+        }
+
+        /// <summary>
+        /// Test adding solver dynamically at runtime to gameobject
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestRuntimeInstantiation()
+        {
+            InstantiateTestSolver<Orbital>();
+
+            yield return null;
+        }
+
+        /// <summary>
+        /// Test solver system's ability to change target types at runtime
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestTargetTypes()
+        {
+            Vector3 rightHandPos = Vector3.right * 50.0f;
+            Vector3 customTransformPos = Vector3.up * 50.0f;
+
+            var transformOverride = new GameObject("Override");
+            transformOverride.transform.position = customTransformPos;
+
+            var testObjects = InstantiateTestSolver<Orbital>();
+
+            // Set solver handler to track hands
+            testObjects.handler.TrackedTargetType = TrackedObjectType.HandJoint;
+
+            InputSimulationService inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+
+            // Test orbital around right hand
+            yield return TestHandSolver(testObjects.target, inputSimulationService, rightHandPos, Handedness.Right);
+
+            // Test orbital around head
+            testObjects.handler.TrackedTargetType = TrackedObjectType.Head;
+
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
+
+            Assert.LessOrEqual(Vector3.Distance(testObjects.target.transform.position, Vector3.zero), DistanceThreshold);
+
```

Comment:
Can you add a comment about where the camera is located (i.e. is it at 2.5m?) so that this check can have a little more context about what it's about

Reply:
Just switched comparison to be the actual head transform in case it changes